synapseformation/create.py

```diff
@@ -28,6 +28,24 @@ def __init__(self, syn: 'Synapse', only_create: bool = True,
         self.logger = logger or logging.getLogger(__name__)
         self._update_str = "Fetched existing" if only_create else "Created"
 
+    def _find_entity_by_name(self, entity_name: str, parentid: str,
+                             concrete_type: str) -> SynapseCls:
+        """Find an Entity by its name
+
+        Args:
+            entity_name: Name of Entity
+            parentid: Synapse parentid
+            concrete_type: Type of Entity
+
+        Returns:
+            Entity
+        """
+        body = json.dumps({"parentId": parentid, "entityName": entity_name})
+        entity_obj = self.syn.restPOST("/entity/child", body=body)
+        new_obj = self.syn.get(entity_obj['id'], downloadFile=False)
+        assert concrete_type == new_obj.properties.concreteType, "Different types."  # pylint: disable=line-too-long
```

Comment:
In case you weren't aware of this trick to have the assert error message span multiple lines:

```suggestion
        assert concrete_type == new_obj.properties.concreteType, (
            f"Retrieved '{entity_name}' had type '{new_obj.properties.concreteType}' "
            f"rather than the expected type '{concrete_type}'."
        )
```

If you disabled pylint here because you like to keep your code "tight", that's totally fine. I just wanted to make sure you knew about the above trick (and you probably already do).